backend/src/test/java/com/pickpick/message/application/MessageServiceQueryTest.java

```diff
@@ -0,0 +1,296 @@
+package com.pickpick.message.application;
+
+import static com.pickpick.fixture.MessageRequestFactory.emptyQueryParams;
+import static com.pickpick.fixture.MessageRequestFactory.emptyQueryParamsWithCount;
+import static com.pickpick.fixture.MessageRequestFactory.fromLatestInChannels;
+import static com.pickpick.fixture.MessageRequestFactory.futureFromTargetMessageInChannels;
+import static com.pickpick.fixture.MessageRequestFactory.pastFromTargetMessageInChannels;
+import static com.pickpick.fixture.MessageRequestFactory.searchByKeywordInChannels;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import com.pickpick.channel.domain.Channel;
+import com.pickpick.channel.domain.ChannelRepository;
+import com.pickpick.channel.domain.ChannelSubscription;
+import com.pickpick.channel.domain.ChannelSubscriptionRepository;
+import com.pickpick.fixture.MessageFixtures;
+import com.pickpick.member.domain.Member;
+import com.pickpick.member.domain.MemberRepository;
+import com.pickpick.message.domain.Message;
+import com.pickpick.message.domain.MessageRepository;
+import com.pickpick.message.ui.dto.MessageRequest;
+import com.pickpick.message.ui.dto.MessageResponse;
+import com.pickpick.message.ui.dto.MessageResponses;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.transaction.annotation.Transactional;
+
+@Sql({"/truncate.sql"})
+@Transactional
+@SpringBootTest
+class MessageServiceQueryTest {
+
+    private static final int MESSAGE_COUNT = 5;
+    private static final int MESSAGE_COUNT_DEFAULT = 20;
+    private static final String MESSAGE_KEYWORD = "줍줍";
+    private static final int TARGET_INDEX = 15;
+    private static final int VIEW_ORDER_FIRST = 1;
+    private static final int VIEW_ORDER_SECOND = 2;
+
+    @Autowired
+    private MemberRepository members;
+
+    @Autowired
+    private ChannelRepository channels;
+
+    @Autowired
+    private ChannelSubscriptionRepository subscriptions;
+
+    @Autowired
+    private MessageRepository messages;
+
+    @Autowired
+    private MessageService messageService;
+
+    @DisplayName("쿼리 파라미터와 count가 없다면, 회원의 첫번째 순서 구독 채널의 최신 메시지를 20개 내림차순 조회한다")
```

Comment:
```suggestion
    @DisplayName("count만 전달됐다면, 회원의 첫번째 순서 구독 채널의 최신 메시지 20개를 작성시간 내림차순 조회한다")
```

내림차순의 기준이 명시되어 있지 않아 제안해봅니다!

Reply:
count는 쿼리 파라미터와는 다르다고 생각하신걸까요 ?-?
저는 count도 쿼리 파라미터도 포함된다고 생각했는데 별도로 명시하셨기에 궁금해서 여쭤봅니다!

Reply:
😅 동일한 쿼리 파람인데 나머지는 필터링/검색의 조건, count는 별도로 자를 개수... 라는 생각이 깔려있어서 이렇게 적은 것 같네요  
수정했어요~